### PR TITLE
Fix creation of resources back links.

### PR DESCRIPTION
**Pull request by DuncanFenning**
This prevents incorrect backlinks being created with many-to-many relationships.

### CODE REVIEW

File: lib/Resource.js [hunk 1]
```diff
@@ -250,18 +250,18 @@ Resource.prototype.delete = Promise.denodeify(function(callback) {
...
-    if (someRawRelationship.meta.as !== relationName) return;
+    if (someRawRelationship.meta.belongsTo !== resource._raw.type) return;
 
-    self["_" + method + "LinksTo"](i, relationName, resource);
+    self["_" + method + "LinksTo"](i, resource, someRawRelationship.meta.many);
   });
 };
 
-Resource.prototype._removeLinksTo = function(i, relationName, resource) {
+Resource.prototype._removeLinksTo = function(i, resource) {
```
Comment (DuncanFenning) [inline lib/Resource.js:264]: relationName was never used so this has been removed

File: lib/Resource.js [hunk 2]
```diff
@@ -273,10 +273,14 @@ Resource.prototype._removeLinksTo = function(i, relationName, resource) {
   }
 };
 
-Resource.prototype._addLinksTo = function(i, relationName, resource) {
+Resource.prototype._addLinksTo = function(i, resource, many) {
```
Comment (DuncanFenning) [inline lib/Resource.js:276]: Again, relationName was an unused param

**Comment by hx-damian-casey** [comment]
:+1: once the tests pass. Code changes make sense and have seen this demo'd successfully.

**Comment by DuncanFenning** [comment]
The tests are failing with CORS issues. We have been merging with failing tests as we have ensured they pass locally

**Comment by msaspence** [comment]
Have yourself an :ant: of approval

**Comment by DuncanFenning** [comment]
We only have Collaborator access, and therefore cannot force merge this PR. We will merge this PR as soon as we have access

**Comment by theninj4** [comment]
@DuncanFenning can you run `npm build` and commit the new dist files - then I'll merge :+1: 

Sorry, I didn't realise you guys can't merge away with the failing CORS tests. I'll tweak the settings.

**Pull request merged**